syntax = "proto3";

package aea.algovera.rabbitmq.v0_1_0;

message RabbitmqMessage{

  // Performatives and contents
  message Consume_Request_Performative{
    map<string, string> rabbitmq_details = 1;
    string consume_queue_name = 2;
  }

  message Consume_Response_Performative{
    map<string, string> consume_response = 1;
  }

  message Publish_Request_Performative{
    map<string, string> rabbitmq_details = 1;
    string publish_queue_name = 2;
    map<string, string> publish_message = 3;
  }

  message Publish_Response_Performative{
    map<string, string> publish_response = 1;
  }


  oneof performative{
    Consume_Request_Performative consume_request = 5;
    Consume_Response_Performative consume_response = 6;
    Publish_Request_Performative publish_request = 7;
    Publish_Response_Performative publish_response = 8;
  }
}
